### PR TITLE
SAK-44629 Dark mode bug fixes

### PR DESCRIPTION
**Pull request by profmikegreene**
https://jira.sakaiproject.org/browse/SAK-44629

This is ready to review but not merge.

I've committed the SWITCH update for the profile tool in-place for easier review and will update that commit, deleting the CSS file and moving it to library/sass before merge.

Not every scss has been reviewed but most have https://docs.google.com/spreadsheets/d/1yf6KGnMxDQIHKZTTRy_iN2fkW2dKAQJzkKTOBeXr31U/edit#gid=1356305061 tracks the status of all the CSS/SCSS files

### CODE REVIEW

File: profile2/tool/src/webapp/css/profile2.css [hunk 1]
```diff
@@ -15,342 +15,327 @@
  */
```
Comment (profmikegreene) [inline profile2/tool/src/webapp/css/profile2.css:15]: This file will be removed when the PR is moved out of draft mode. These changes have been moved into profil2.scss under library. Keeping them here makes the review easier to scan and understand.

File: library/src/morpheus-master/sass/modules/tool/profile2/_profile2.scss [hunk 2]
```diff
@@ -1,10 +1,1224 @@
 .#{$namespace}sakai-profile2 {
 
+	.profile-twitter-account td.label{
```
Comment (profmikegreene) [inline library/src/morpheus-master/sass/modules/tool/profile2/_profile2.scss:3]: I don't recommend reviewing the first portion file, I've left the original profile2.css file in this commit that contains the same changes as lines 3-1216 of this file as that is easier to see the difference in the refactor.

Lines 1217-1344 of this file are new changes that should be reviewed and are not contained elsewhere. I'll remove the old profile2.css file before merge.

File: library/src/morpheus-master/sass/modules/connection-manager/_base.scss [hunk 3]
```diff
@@ -240,10 +238,10 @@
       .connection-manager-social-icon {
         font-size: 20px;
         margin-right: 10px;
-        color: #666666;
+        color: var(--sakai-text-color-2);
       }
       .inactive {
-        color: #A9A9A9 !important;
+        color: var(--sakai-text-color-disabled) !important;
```
Comment (fostersdesign) [inline library/src/morpheus-master/sass/modules/connection-manager/_base.scss:244]: Is the !important still needed here?

File: library/src/morpheus-master/sass/modules/grader/_base.scss [hunk 4]
```diff
@@ -89,16 +89,15 @@ sakai-grader {
...
-            background: white;
+            background: var(--sakai-background-color-1);
           }
           .preview-inner {
             width: 75%;
-            background: white;
+            background: var(--sakai-background-color-1);
             margin-top: 20px;
             margin-bottom: 20px;
             display: inline-block;
```
Comment (fostersdesign) [inline library/src/morpheus-master/sass/modules/grader/_base.scss:103]: This is outside of the scope of this PR, but I just noticed this here that this padding is missing a unit of pixels for its values.

File: library/src/morpheus-master/sass/modules/portal-search/_base.scss [hunk 5]
```diff
@@ -27,21 +27,21 @@
...
     text-decoration: none;
   }
 
   .search-result-container {
 
     padding-bottom: 9px;
-    border-bottom: 1px solid black;
+    border-bottom: 1px solid var(--sakai-border-color);
 
     text-align: left;
```
Comment (fostersdesign) [inline library/src/morpheus-master/sass/modules/portal-search/_base.scss:46]: A few extra empty lines can be removed here.

File: library/src/morpheus-master/sass/modules/tool/_base.scss [hunk 6]
```diff
@@ -82,14 +82,22 @@
...
 				}
+				&:active, &:focus, &:hover {
+					&:before {
+						color:var(--sakai-color-red--darker-2);
+					}
+				}
+				&:before {
+					@extend .fa;
+					color:var(--sakai-color-red--darker-2);
+				}
```
Comment (fostersdesign) [inline library/src/morpheus-master/sass/modules/tool/_base.scss:100]: Normal text colours here are probably okay. This doesn't need to be a red "X" in my opinion.

File: library/src/morpheus-master/sass/modules/tool/assignments/_assignments.scss [hunk 7]
```diff
@@ -412,77 +417,77 @@
...
 		@extend .contentReviewIcon;
-		color: #FFD600;
+		color: var(--warnBanner-color);
 	}
 
 	.contentReviewIconThreshold-3-draft {
 		@extend .contentReviewIconDraft;
-		color: #FFD600;
+		color: var(--warnBanner-color);
 	}
```
Comment (fostersdesign) [inline library/src/morpheus-master/sass/modules/tool/assignments/_assignments.scss:461]: We might need more colours for these six levels to better illustrate the various threshold levels.

File: library/src/morpheus-master/sass/modules/tool/commons/_commons.scss [hunk 8]
```diff
@@ -216,11 +216,7 @@
             font-size: smaller;
         }
 
-    .qtip.commons-qtip {
-        border: 1px #E0E0E0 solid !important;
-        background-color: #F8F8F8 !important;
-        max-width: 310px;
-    }
+
```
Comment (fostersdesign) [inline library/src/morpheus-master/sass/modules/tool/commons/_commons.scss:220]: Couple extra spaces that can be remove to increase your deleted lines count! ;)

**Review by fostersdesign — COMMENTED** [review]
Lots of great work here, @profmikegreene ! I've left a few comments. I still have a few a dozen more files to go through tomorrow. I did jump to profile2.css and reviewed it. Looks good. If the scss is the same, I think you can proceed with the removal of the css file.

File: library/src/morpheus-master/sass/modules/tool/resources/_resources.scss [hunk 9]
```diff
@@ -53,7 +53,7 @@
 			}
 			&.inactive {
 				* {
-					color: #333 !important;
+					color: var(--sakai-text-color-2) !important;
```
Comment (fostersdesign) [inline library/src/morpheus-master/sass/modules/tool/resources/_resources.scss:56]: Do we want to use the disabled colour here?

File: library/src/morpheus-master/sass/modules/tool/roster/_roster.scss [hunk 10]
```diff
@@ -119,20 +119,22 @@
                         width: 32px;
                         height: 32px;
                         border-radius: 16px;
-                        background-color: #ddd;
-                        border: 1px solid #bbb;
+                        background-color: var(--sakai-background-color-3);
+                        border: 2px solid var(--sakai-border-color);
```
Comment (fostersdesign) [inline library/src/morpheus-master/sass/modules/tool/roster/_roster.scss:123]: Was the extra pixel intentional?

File: library/src/morpheus-master/sass/modules/tool/sitestats/_sitestats.scss [hunk 11]
```diff
@@ -215,469 +221,399 @@
...
-    -moz-border-radius-topright:4px;
-    -webkit-border-top-left-radius:4px;
-    -webkit-border-top-right-radius:4px;
-    /*behavior: url(/sitestats-tool/script/border-radius.htc);*/
+    text-decoration: none;
+    border:1px solid var(--sakai-border-color);
+    background: var(--sakai-background-color-1);
     border-top-left-radius:4px;
     border-top-right-radius:4px;
+    padding-top: 5px; /* with respect to padding (X) above {ul.tabnav} and below {ul.tabnav li a} */
```
Comment (fostersdesign) [inline library/src/morpheus-master/sass/modules/tool/sitestats/_sitestats.scss:300]: Should just include this padding in the padding defined higher in the rule.

File: library/src/morpheus-master/sass/modules/tool/sitestats/_sitestats.scss [hunk 12]
```diff
@@ -215,469 +221,399 @@
...
   }
 
-
-  /** Ajax loading */
-  .ajaxLoading {
-    background: #ffffff url(/sitestats-tool/images/ajaxload.gif) no-repeat;
-    color: window;
-  }
-
-
```
Comment (fostersdesign) [inline library/src/morpheus-master/sass/modules/tool/sitestats/_sitestats.scss:467]: Is this not needed?

File: library/src/morpheus-master/sass/modules/tool/sitestats/_sitestats.scss [hunk 13]
```diff
@@ -215,469 +221,399 @@
...
-    background-repeat: no-repeat;
-    background-position:right center;
-    padding-right: 16px;
-  }
-  .wicket_orderDown a span {
-    background-image: url('/library/image/sakai/sortdescending.gif');
-    background-repeat: no-repeat;
-    background-position: right center;
-    padding-right: 16px;
-  }
```
Comment (fostersdesign) [inline library/src/morpheus-master/sass/modules/tool/sitestats/_sitestats.scss:519]: This is needed by Wicket.

File: library/src/morpheus-master/sass/modules/tool/sitestats/_sitestats.scss [hunk 14]
```diff
@@ -215,469 +221,399 @@
...
-    display: inline-block;
-    height: 15px;
-    width: 100%;
-    border: 1px solid #777;
-    text-align: center;
-    vertical-align: middle;
-    font-family:Arial,Helvetica,sans-serif;
-    font-size: 11px;
-    font-weight: bold;
-    color: #000;
```
Comment (fostersdesign) [inline library/src/morpheus-master/sass/modules/tool/sitestats/_sitestats.scss:534]: Is this needed for the graphs?

File: library/src/morpheus-master/sass/modules/tool/syllabus/_syllabus.scss [hunk 15]
```diff
@@ -51,11 +51,11 @@
 
   .ui-accordion .ui-accordion-header {
     overflow: auto !important;
-    background-color: #D9EEF7;
+    background-color: var(--infoBanner-bgcolor);
```
Comment (fostersdesign) [inline library/src/morpheus-master/sass/modules/tool/syllabus/_syllabus.scss:54]: I'd like to see these a more generic colour, like grey. But maybe this could be a different PR.

File: library/src/webapp/skin/tool_base.css [hunk 16]
```diff
@@ -235,7 +229,7 @@ ul.attachList li{
 /*not sure where used - samigo?*/
 span.abc{
 	letter-spacing: .2em;
-	color: #555;
+	color: var(--sakai-background-color-2);
```
Comment (fostersdesign) [inline library/src/webapp/skin/tool_base.css:232]: Do you want a text colour here?

File: library/src/webapp/skin/tool_base.css [hunk 17]
```diff
@@ -291,7 +285,7 @@ table.discTria img, .topicHeadings img,  .msgHeadings img{
 	font-family: verdana,arial;
```
Comment (fostersdesign) [inline library/src/webapp/skin/tool_base.css:285]: Let's get rid of this font-family, please.

File: library/src/webapp/skin/tool_base.css [hunk 18]
```diff
@@ -835,8 +808,8 @@ blockquote p {
 .ui-tpicker-grid-label { background: none; border: none; margin: 0; padding: 0; }
 .ui-timepicker-div .ui_tpicker_unit_hide{ display: none; }
 
-.ui-timepicker-div .ui_tpicker_time .ui_tpicker_time_input { background: none; color: inherit; border: none; outline: none; border-bottom: solid 1px #555; width: 95%; }
-.ui-timepicker-div .ui_tpicker_time .ui_tpicker_time_input:focus { border-bottom-color: #aaa; }
+.ui-timepicker-div .ui_tpicker_time .ui_tpicker_time_input { background: none; color: inherit; border: none; outline: none; border-bottom: solid 1px var(--sakai-border-color); width: 95%; }
+.ui-timepicker-div .ui_tpicker_time .ui_tpicker_time_input:focus { border-bottom-color: var(--sakai-border-color); }
```
Comment (fostersdesign) [inline library/src/webapp/skin/tool_base.css:812]: Do we need a different border colour here to distinguish the focus state change?

File: library/src/morpheus-master/sass/modules/tool/roster/_roster.scss (hunk 10 above)
Comment (profmikegreene) [reply library/src/morpheus-master/sass/modules/tool/roster/_roster.scss:123]: yes

File: library/src/morpheus-master/sass/modules/tool/resources/_resources.scss (hunk 9 above)
Comment (profmikegreene) [reply library/src/morpheus-master/sass/modules/tool/resources/_resources.scss:56]: can do

File: library/src/morpheus-master/sass/modules/tool/sitestats/_sitestats.scss (hunk 12 above)
Comment (profmikegreene) [reply library/src/morpheus-master/sass/modules/tool/sitestats/_sitestats.scss:467]: I didn't think so but Shawn mentioned it might be needed by wicket elsewhere, so i'm putting it back in.

File: library/src/morpheus-master/sass/modules/tool/sitestats/_sitestats.scss (hunk 13 above)
Comment (profmikegreene) [reply library/src/morpheus-master/sass/modules/tool/sitestats/_sitestats.scss:519]: Putting them back in.

File: library/src/morpheus-master/sass/modules/tool/sitestats/_sitestats.scss (hunk 14 above)
Comment (profmikegreene) [reply library/src/morpheus-master/sass/modules/tool/sitestats/_sitestats.scss:534]: I don't think so, the charts seem to not use any of these classes from what i can tell.

File: library/src/morpheus-master/sass/modules/tool/syllabus/_syllabus.scss (hunk 15 above)
Comment (profmikegreene) [reply library/src/morpheus-master/sass/modules/tool/syllabus/_syllabus.scss:54]: this is overridden later in the file anyway so i'll delete this rule.

**Review by fostersdesign — APPROVED** [review]
Thanks for making all of those changes, @profmikegreene! Looks good to me.
